docs/content/python/new-roles/getting-started.md

```diff
@@ -0,0 +1,473 @@
+  roles for this resource.
+
+In our rule body, we first define the list of available actions for this
+resource:
+
+```polar
+resource(_type: Org, "org", actions, roles) if
+    actions = ["read", "create_repo"] and
+    roles = {
+        ...
+    };
+```
+
+Now, we define our roles. Roles are defined in a dictionary that maps
+the role name to a role configuration.
+
+```polar
+resource(_type: Org, "org", actions, roles) if
+    actions = ["read", "create_repo"] and
+    roles = {
+        org_member: {
+            perms: ["read"],
+        },
+        org_owner: {
+            perms: ["read", "create_repo"],
+        }
+    };
+```
+
+This resource definition defines two roles:
+
+- *org_member*: Has the `read` permission.
+- *org_owner*: Has the `read` and `create_repo` permissions.
+
+Permissions are actions associated with a resource type. A permission can
+directly reference an action defined in the same resource. Later, we'll
+see how to add permissions defined on other resources to roles.
+
+{{% callout "resource(...) is just a rule" "blue" %}}
+
+The `resource` definition is just a regular Polar rule. That's why it
+has an `if` and `and` between variable assignments. `actions` and
+`roles` are unbound parameters, meaning they can be assigned inside of
+the rule body.
+
+We could have written this rule without a body, like:
+
+```polar
+resource( _type: Org, "org", ["read", "create_repo"],
+    {
+        org_member: {
+            perms: ["read"],
+        },
+        org_owner: {
+            perms: ["read", "create_repo"],
+        }
+    }
+);
+```
+
+But, we think the expanded form is clearer.
+
+{{% /callout %}}
+
+### Adding role_allow to our policy
+
+To allow access based on roles, we call `Roles.role_allow` in our
+policy:
+
+{{< literalInclude
+    path="examples/gitclub-sqlalchemy-flask-react/backend/app/authorization.polar"
+    from="docs: begin-role-allow"
+    to="docs: end-role-allow"
+    gitHub="https://github.com/osohq/gitclub-sqlalchemy-flask-react"
+    linenos=true
+>}}
+
+Oso will now allow access to any resource that is allowed based on the
+role definitions.
+
+### Assigning roles to users
+
+Now we've configured roles and setup our policy. For users to have
+access, we must assign them to roles.
+
+{{< literalInclude
+    path="examples/gitclub-sqlalchemy-flask-react/backend/app/routes.py"
+    from="docs: begin-role-assignment"
+    to="docs: end-role-assignment"
+    gitHub="https://github.com/osohq/gitclub-sqlalchemy-flask-react"
+    linenos=true
+    >}}
+
+**TODO** api docs
+
+The `assign_role` function assigns a particular role on a resource.
+Often, you'll make this available in a route so that end users can
+configure role assignments for their organization. More on this in [end
+user configuration](end_user_configuration).
+
+Role assignment is stored your database along with the rest of your
+SQLAlchemy managed data.
+
+### Persisting roles configuration
```

Comment:
This kind of feels like a random place to put this, would have expected it to be with the initialization.